docs/source/60_building_apps.md

```diff
@@ -1,3 +1,250 @@
 # Building Downstream Applications {#apps}
 
-ADAM is packaged so that it can be used via the base CLI, to run plugins via the CLI, or as a library which can be used when building downstream applications. This document covers some of the important things to note when building applications downstream of ADAM.
+ADAM is packaged so that it can be used interatively via the ADAM shell, called from
+the command line interface (CLI), or included as a library when building downstream
+applications.
+
+This document covers three patterns for building applications downstream of ADAM:
+
+* Extend the ADAM CLI by [adding new commands](#commands)
+* Extend the ADAM CLI by [adding new commands in an external repository](#external-commands)
+* Use ADAM as a [library in new applications](#library)
+
+
+## Extend the ADAM CLI by adding new commands {#commands}
+
+ADAM's CLI is implemented in the adam-cli Apache Maven module of the
+[bdgenomics/adam](https://github.com/bigdatagenomics/adam) repository, one
+.scala source file for each CLI action (e.g. [Transform.scala](https://github.com/bigdatagenomics/adam/blob/master/adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala)
+for the [transform](#transform) action), and a main class ([ADAMMain.scala](https://github.com/bigdatagenomics/adam/blob/master/adam-cli/src/main/scala/org/bdgenomics/adam/cli/ADAMMain.scala))
+that assembles and delegates to the various CLI actions.
+
+To add a new command:
+
+Extend `Args4jBase` class to specify arguments to the command. Arguments are defined using
+the [args4j library](http://args4j.kohsuke.org/). If reading or writing to Parquet, consider
+including Parquet arguments via `with ParquetArgs`.
+
+```scala
+class MyCommandArgs extends Args4jBase with ParquetArgs {
+  @Argument(required = true, metaVar = "INPUT", usage = "Input to my command", index = 0)
+  var inputPath: String = null
+}
+```
+
+Extend `BDGCommandCompanion` object to specify the command name and description. The `apply`
+method associates `MyCommandArgs` defined above with `MyCommand`.
+
+```scala
+object MyCommand extends BDGCommandCompanion {
+  val commandName = "myCommand"
+  val commandDescription = "My command example."
+
+  def apply(cmdLine: Array[String]) = {
+    new MyCommand(Args4j[MyCommandArgs](cmdLine))
+  }
+}
+```
+
+Extend `BDGSparkCommand` class and implement the `run(SparkContext)` method. The `MyCommandArgs`
+class defined above is provided in the constructor and specifies the generic type for `BDGSparkCommand`.
+The companion object defined above is declared as a field.  For access to the Apache Spark
+[slf4j](http://www.slf4j.org/) Logger via the `log` field, specify `with Logging`.
```

Comment:
Small nit: not the Spark logger anymore, since they moved that to be package private. It's the BDG utils logger now.